### ISSUE TITLE
'Revoke' user invite requests not working on `/team/`

### ISSUE THREAD

**Issue by ErisDS** [TryGhost/Ghost#5947]
When attempting to revoke an invite to a user on the team page, nothing happens.

It seems that the `status=all` part of the request is missing in the call to the users API endpoint. As the users API will only return `active` users by default, the user will not be found, and a 404 is returned instead.

### Steps to Reproduce

1. On `/team/`, invite a user to your blog
2. Hit the 'revoke' button, nothing happens

The API returns a 404: 

![](http://puu.sh/kK5Bc.png)

The console shows an error:

![](http://puu.sh/kK5JZ.png)

### Technical details: 

* Ghost Version: master

**Comment by kevinansfield** [comment]
I've submitted a quick-fix in #5953 but I think the API behaviour is wrong in this case - _querying by ID_ should by default query all available records (based on permissions, but that's not ready yet).

If there's a use case where you want to restrict the the query-able record set then that should be indicated by specifying a modifier such as `status=invited`.

For _index queries_ I think the current behaviour makes sense as there you want to provide sensible defaults and then expand/restrict the result set with query modifiers.

**Comment by ErisDS** [comment]
I fully agree with the assessment here, it's another subtle detail of how different requests need to behave very differently. 

I've added the permissions label here so it's tracked as part of the new permissions spec. There is an additional complexity here in determining when we should return 403 or 404 - as 403 gives away that a resource exists?

**Comment by kevinansfield** [comment]
> There is an additional complexity here in determining when we should return 403 or 404 - as 403 gives away that a resource exists?

Does giving away that a resource exists matter as long as it's only accessible with the correct permissions? I'd err on the side of sticking to the HTTP spec, a client would be able to provide a more relevant error message if it received a 403, plus the client already expected something at that location anyway and would need to be successfully authenticated else it would be a 401?

**Comment by sebgie** [comment]
Ok, that's an interesting question.

> I think the API behaviour is wrong in this case - querying by ID should by default query all available records (based on permissions, but that's not ready yet?).

That would be a very sensible change in my opinion. The current `status=all` came from the initial implementation and it would probably be easier to understand if we would apply a filter to limit the result rather than having a filter to show all results?

>403 vs 404

In my opinion this depends on the resources the authenticated user is allowed to see. If the user is allowed to view the resource, but doesn't have permission to edit the resource the request to edit returns 403. If the user is not allowed to see the resource all requests return 404.

Example, user viewable but no permission:

- authenticated: yes/no
    - yes: proceed
    - no: 401 unauthorized
- GET /users/3
    - 200: JSON response
- PUT /users/3
    - 403 forbidden

Example, user not viewable, no permission

- authenticated: yes/no
    - yes: proceed
    - no: 401 unauthorized
- GET /users/3
    - 404 not found
- PUT /users/3
    - 404 not found

Returning 403 for all existing resources would be similar to showing all users that are in the database in the UI to any user without information. It would simply be a placeholder to tell the user there is something you will never be able to access.

This also takes into account various security considerations about leaking information by confirming that a certain resource exists.

**Comment by ErisDS** [comment]
The current work to split models should resolve this issue.

**Comment by gargol** [comment]
The underlying issue has been fixed time ago when according enforced/default filters were introduced on the model layer:https://github.com/TryGhost/Ghost/blob/4acc375fb600bf9b46041e26e3d06b71a5b141b1/core/server/models/user.js#L293

Model separation is a separate improvement that could be done on model layer, but does not directly touch current bug.
